controllers/drplacementcontrol_controller.go

```diff
@@ -699,6 +699,11 @@ func (d *DRPCInstance) runInitialDeployment() (bool, error) {
 
 	const done = true
 
+	if !d.isValidInitialDeploymentState() {
```

Comment:
(not a comment on the PR, but the CI failure) This will still not fix the problem in CI, as status can read "Deployed" and we would still continue with initial placement.

Reply:
This is done in line 727 below, I was looking at it one commit at a time.